kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/mwp/MwpNewsProvider.java

```diff
@@ -0,0 +1,333 @@
+    URI uri;
+    
+    try {
+      uri = new URIBuilder(imageUrl).build();
+    } catch (URISyntaxException e) {
+      logger.log(Level.SEVERE, String.format("Invalid uri %s", imageUrl), e);
+      return null;
+    }
+    
+    return getImageData(uri);
+  }
+
+  private AttachmentData getImageData(URI uri) {
+    Response<BinaryResponse> response = binaryHttpClient.downloadBinary(uri);
+    if (response.isOk()) {
+      return new AttachmentData(response.getResponseEntity().getType(), response.getResponseEntity().getData());
+    } else {
+      logger.severe(String.format("Image download failed on [%d] %s", response.getStatus(), response.getMessage()));
+    }
+    
+    return null;
+  }
+  
+  private AttachmentId getImageAttachmentId(Integer id) {
+    AttachmentId mwpId = new AttachmentId(MwpConsts.IDENTIFIER_NAME, String.valueOf(id));
+    AttachmentId kuntaApiId = idController.translateAttachmentId(mwpId, KuntaApiConsts.IDENTIFIER_NAME);
+    
+    if (kuntaApiId == null) {
+      logger.info(String.format("Found new mwp attachment %d", id));
+      Identifier newIdentifier = identifierController.createIdentifier(mwpId);
+      kuntaApiId = new AttachmentId(KuntaApiConsts.IDENTIFIER_NAME, newIdentifier.getKuntaApiId());
+    }
+    
+    return kuntaApiId;
+  }
+  
+  private Integer getMediaId(AttachmentId attachmentId) {
+    AttachmentId mwpAttachmentId = idController.translateAttachmentId(attachmentId, MwpConsts.IDENTIFIER_NAME);
+    if (mwpAttachmentId == null) {
+      logger.info(String.format("Could not translate %s into mwp ", attachmentId.toString()));
+      return null;
+    }
+    
+    return NumberUtils.createInteger(mwpAttachmentId.getId());
+  }
+
+  private fi.otavanopisto.mwp.client.model.Attachment findMedia(Integer featuredMediaId) {
+    if (featuredMediaId == null) {
+      return null;
+    }
+    
+    ApiResponse<fi.otavanopisto.mwp.client.model.Attachment> response = mwpApi.getApi().wpV2MediaIdGet(String.valueOf(featuredMediaId), null);
+    if (!response.isOk()) {
+      logger.severe(String.format("Finding media failed on [%d] %s", response.getStatus(), response.getMessage()));
+    } else {
+      return response.getResponse();
+    }
+    
+    return null;
+  }
+
+  private Post findPostByArticleId(NewsArticleId newsArticleId) {
+    NewsArticleId kuntaApiId = idController.translateNewsArticleId(newsArticleId, MwpConsts.IDENTIFIER_NAME);
+    if (kuntaApiId == null) {
+      logger.severe(String.format("Failed to convert %s into MWP id", newsArticleId.toString()));
+      return null;
+    }
+    
+    ApiResponse<Post> response = mwpApi.getApi().wpV2PostsIdGet(kuntaApiId.getId(), null);
+    if (!response.isOk()) {
+      logger.severe(String.format("Finding post failed on [%d] %s", response.getStatus(), response.getMessage()));
+    } else {
+      return response.getResponse();
+    }
+    
+    return null;
+  }
+
+  private List<NewsArticle> translateNewsArticles(List<Post> posts) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateNewsArticles" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)